lib/fprotect/fprotect_acl.c

```diff
@@ -16,9 +16,15 @@ int fprotect_area(uint32_t start, size_t length)
 	__ASSERT_NO_MSG(nrf_ficr_codepagesize_get(NRF_FICR) ==
 			CONFIG_FPROTECT_BLOCK_SIZE);
 
+#if defined(FPROTECT_RESERVE_HUK_ACL_SLOT)
```

Comment:
Why is the fprotect modification needed?

Reply:
I don't see that it hurts, as fprotect can be used in other scopes... What do you propose as an alternative to assured reservation of the last ACL slot?

Reply:
I get it now, but I do see possibilities for it going wrong, if the region index is changed in one place, or another module needs a region. IMO it would be better to modify fprotect so the HUK page protection can be done through fprotect. Alternatively, add a function in fprotect for getting a region.

Reply:
I also thought about extending fprotect before but since fprotect  has multiple backends I thought this is the less invasive change. Modifying fprotect to get an available region could be an option of course but it will require some (short) slot management code addition and some header file changes only for fprotect with acl. Could clarify a bit what do you mean with "the region index can be changed in one place". I am not sure that I got the point. Since we just remote one slot from the managment of the fprotect, how could it get changed?

Reply:
>"the region index can be changed in one place". I am not sure that I got the point

The actual index (right now 7) could be changed. And since writes to ACL configuration registers are ignored when already written to, this error would not be noticed, but the page would no longer be protected.

>Modifying fprotect to get an available region could be an option of course but it will require some (short) slot management code addition and some header file changes only for fprotect with acl.

The management code is already there as `region_idx++` and

    if (region_idx >= ACL_REGIONS_COUNT) {
        return -ENOSPC;
    }

it just needs to be extracted into a function.

Reply:
That sounds very reasonable indeed! It sounds risky to have a possibility for this to fail without noticing.